src/main/java/samza/examples/cookbook/CouchbaseTableExample.java

```diff
@@ -0,0 +1,259 @@
+ *     2019-05-23 21:18:07 2019-05-23 21:18:07 word=3, count=1, total-count=4
+ *     2019-05-23 21:18:08 2019-05-23 21:18:08 word=4, count=2, total-count=5
+ *     2019-05-23 21:18:08 2019-05-23 21:18:08 word=5, count=1, total-count=6
+ *     2019-05-23 21:18:09 2019-05-23 21:18:09 word=2, count=2, total-count=7
+ *     2019-05-23 21:18:09 2019-05-23 21:18:09 word=3, count=2, total-count=8
+ *     2019-05-23 21:18:10 2019-05-23 21:18:10 word=1, count=2, total-count=9
+ *
+ *     You can examine the result on Couchbase Admin GUI as well.
+ *
+ *     Note:
+ *       - If you enter "1 2 3 4 5 4 3 2 1", you should see roughly 1 QPS as
+ *         the input is processed by only one task
+ *
+ *
+ *   </li>
+ * </ol>
+ *
+ */
+public class CouchbaseTableExample implements StreamApplication {
+
+  private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+  private static final String KAFKA_SYSTEM_NAME = "kafka";
+  private static final List<String> KAFKA_CONSUMER_ZK_CONNECT = ImmutableList.of("localhost:2181");
+  private static final List<String> KAFKA_PRODUCER_BOOTSTRAP_SERVERS = ImmutableList.of("localhost:9092");
+  private static final Map<String, String> KAFKA_DEFAULT_STREAM_CONFIGS = ImmutableMap.of("replication.factor", "1");
+
+  private static final String INPUT_STREAM_ID = "word-input";
+  private static final String OUTPUT_STREAM_ID = "count-output";
+
+  private static final String CLUSTER_NODES = "couchbase://127.0.0.1";
+  private static final int COUCHBASE_PORT = 11210;
+  private static final String BUCKET_NAME = "my-bucket";
+  private static final String BUCKET_PASSWORD = "123456";
+  private static final String TOTAL_COUNT_ID = "total-count";
+
+  @Override
+  public void describe(StreamApplicationDescriptor app) {
+
+    KafkaSystemDescriptor kafkaSystemDescriptor = new KafkaSystemDescriptor(KAFKA_SYSTEM_NAME)
+        .withConsumerZkConnect(KAFKA_CONSUMER_ZK_CONNECT)
+        .withProducerBootstrapServers(KAFKA_PRODUCER_BOOTSTRAP_SERVERS)
+        .withDefaultStreamConfigs(KAFKA_DEFAULT_STREAM_CONFIGS);
+
+    KafkaInputDescriptor<String> wordInputDescriptor =
+        kafkaSystemDescriptor.getInputDescriptor(INPUT_STREAM_ID, new StringSerde());
+
+    KafkaOutputDescriptor<String> countOutputDescriptor =
+        kafkaSystemDescriptor.getOutputDescriptor(OUTPUT_STREAM_ID, new StringSerde());
+
+    MyCouchbaseTableWriteFunction writeFn = new MyCouchbaseTableWriteFunction(BUCKET_NAME, CLUSTER_NODES)
+        .withBootstrapCarrierDirectPort(COUCHBASE_PORT)
+        .withUsernameAndPassword(BUCKET_NAME, BUCKET_PASSWORD)
+        .withTimeout(Duration.ofSeconds(5));
+
+    TableRetryPolicy retryPolicy = new TableRetryPolicy()
+        .withFixedBackoff(Duration.ofSeconds(1))
+        .withStopAfterAttempts(3);
+
+    RemoteTableDescriptor couchbaseTableDescriptor = new RemoteTableDescriptor("couchbase-table")
+        .withReadFunction(new DummyTableReadFunction())
```

Comment:
Is there any reason we don't allow a write only table descriptor?

Reply:
No really, it was perceived a table should always be readable.